### PR TITLE
[#133] Patching EDSL for arrays of tables

### LINKED ISSUE #133: Road to Arrays of Tables - Patching EDSL
Patch Edsl to allow creating of array of tables.

### PR DESCRIPTION
**Pull request by jiegillet**
Resolves #133 
### :white_check_mark: Check list
- [x] CHANGELOG
- [x] New/fixed features work as expected (No new tests).
- [x] There are no warnings during compilation.
- [x] `hlint .` output is: _No Hints_ .
- [x] The code is formatted with the `stylish-haskell` tool 
      using [stylish-haskell.yaml][stylish] file in the repository.
- [x] The code style of the files you changed is preserved.
- [ ] Commit messages are in the proper format.
      Start the first line of the commit with the issue number in square parentheses.

### CODE REVIEW

File: src/Toml/Edsl.hs [hunk 1]
```diff
@@ -31,10 +39,18 @@ type TDSL = State TOML ()
...
 (=:) :: Key -> Value a -> TDSL
 (=:) k v = modify $ insertKeyVal k v
 
 -- | Adds table to the 'TDSL'.
 table :: Key -> TDSL -> TDSL
 table k = modify . insertTable k . mkToml
+
+-- | Adds array of tables to the 'TDSL'.
+array :: Key -> NonEmpty TDSL -> TDSL
+array k = modify . insertTableArrays k . fmap mkToml
```
Comment (ChShersh) [inline src/Toml/Edsl.hs]: I cant' think of better API for inserting array of tables, so let's keep this type and this implementation.

File: src/Toml/Edsl.hs [hunk 2]
```diff
@@ -31,10 +39,18 @@ type TDSL = State TOML ()
...
 -- | Adds key-value pair to the 'TDSL'.
 (=:) :: Key -> Value a -> TDSL
 (=:) k v = modify $ insertKeyVal k v
 
 -- | Adds table to the 'TDSL'.
 table :: Key -> TDSL -> TDSL
 table k = modify . insertTable k . mkToml
+
+-- | Adds array of tables to the 'TDSL'.
+array :: Key -> NonEmpty TDSL -> TDSL
```
Comment (ChShersh) [inline src/Toml/Edsl.hs]: I feel like `array` is not the best name because it conflicts with ordinary arrays in TOML.

Comment (jiegillet) [reply src/Toml/Edsl.hs]: `tableArray`?

File: src/Toml/Edsl.hs [hunk 3]
```diff
@@ -9,20 +9,28 @@ exampleToml = mkToml $ do
     \"key2\" =: Bool True
     table \"tableName\" $
         \"tableKey\" =: Array [\"Oh\", \"Hi\", \"Mark\"]
+    array \"arrayName\" $
```
Comment (vrom911) [inline src/Toml/Edsl.hs]: I guess, it's also should be `tableArray` now?

Comment (jiegillet) [reply src/Toml/Edsl.hs]: Of course, thank you!

**Pull request merged**